test/Chainweb/Test/Pact/PactExec.hs

```diff
@@ -87,154 +201,61 @@ pactTestSetup = do
-parseText (Object o) =
-  case HM.lookup "data" o of
-    Nothing -> Nothing
-    Just (String t) -> Just t
-    Just _ -> Nothing
-parseText _ = Nothing
-
-fileCompareTxLogs :: FilePath -> TestResponse -> IO TestTree
-fileCompareTxLogs fp resp =
-    return $ goldenVsString (takeBaseName fp) (testPactFilesDir ++ fp) ioBs
-    where
-        ioBs = return $ toS $ Y.encode <$> _flTxLogs $ _trOutput resp
-
-----------------------------------------------------------------------------------------------------
--- Pact test datatypes
-----------------------------------------------------------------------------------------------------
-
-data TestRequest = TestRequest
-    { _trCmd :: TestSource
-    , _trEval :: TestResponse -> IO TestTree
-    , _trDisplayStr :: String
-    }
-
-data TestSource = File FilePath | Code String
-  deriving Show
-
-data TestResponse = TestResponse
-    { _trRequest :: TestRequest
-    , _trOutput :: FullLogTxOutput
-    }
-
-instance Show TestRequest where
-    show tr = "cmd: " ++ show (_trCmd tr) ++ "\nDisplay string: "
-              ++ show (_trDisplayStr tr)
-
-instance Show TestResponse where
-    show tr =
-        let tOutput = _trOutput tr
-            cmdResultStr = show $ _flCommandResult tOutput
-            txLogsStr = unlines $ fmap show (_flTxLogs tOutput)
-        in "\n\nCommandResult: " ++ cmdResultStr ++ "\n\n"
-           ++ "TxLogs: " ++ txLogsStr
-
-data PactTestSetup = PactTestSetup
-  { _pactServiceEnv :: PactServiceEnv
-  , _pactDbState :: PactDbState
-  }
-
-----------------------------------------------------------------------------------------------------
--- sample data
-----------------------------------------------------------------------------------------------------
-
-testPactRequests :: Vector TestRequest
-testPactRequests = V.fromList
-    [ testReq1
-    , testReq2
-    , testReq3
-    , testReq4
-    , testReq5
-    ]
-
-testReq1 :: TestRequest
-testReq1 = TestRequest
-    { _trCmd = Code "(+ 1 1)"
-    , _trEval = return . testCase "addition" . checkScientific (scientific 2 0)
-    , _trDisplayStr = "Executes 1 + 1 in Pact and returns 2.0" }
-
-testReq2 :: TestRequest
-testReq2 = TestRequest
-    { _trCmd = File "test1.pact"
-    , _trEval = return . testCase "load module" . checkSuccessOnly
-    , _trDisplayStr = "Loads a pact module" }
-
-testReq3 :: TestRequest
-testReq3 = TestRequest
-    { _trCmd = Code "(create-table test1.accounts)"
-    , _trEval = fileCompareTxLogs "create-table-expected.txt"
-    , _trDisplayStr = "Creates tables" }
-
-testReq4 :: TestRequest
-testReq4 = TestRequest
-    { _trCmd = Code "(test1.create-global-accounts)"
-    , _trEval = fileCompareTxLogs "create-accounts-expected.txt"
-    , _trDisplayStr = "Creates two accounts" }
+-- | A test runner for golden tests.
+--
+fileCompareTxLogs :: String -> IO TestResponse -> ScheduledTest
+fileCompareTxLogs label respIO = pactGoldenSch label $ do
+    resp <- respIO
+    return $ toS $ Y.encode
```

Comment:
Usage of `Y.encode` is problematic. There is no standardized normal form for Yaml or JSON encoded data and the encoding may change between different version of `aeson`/`yaml`, any dependency, `ghc`, platform, etc.

If we find the output to be sufficiently stable it may be fine for golden tests. Otherwise we should implement a normalized JSON encoding.